develop/tutorials/articles/192-item-selector/03-customizing-item-selector-views.markdown

```diff
@@ -0,0 +1,390 @@
+    `itemSelectorCriterion` in your `*ItemSelectorView` class with the
+    `*ItemSelectorCriterion` class you wish to use, such as
+    `BlogsItemSelectorCriterion`.
+
+    Note that the OSGi component is registered with the property
+    `item.selector.view.order`. The Item Selector view order (the order of the
+    rendered tab views) is prioritized according to a couple settings.
+
+    First, it considers the order of the criteria decided by the client of the
+    Item Selector. When the application invokes the Item Selector it will
+    specify the criteria ordered by importance. Second, if there are multiple
+    views for the same criteria, it will use the property
+    `item.selector.view.order` to determine which will appear first. The lower
+    the value is, the more priority it has, and the sooner it will appear in the
+    order.
+
+2.  Next, you'll need to specify which `*ItemSelectorCriterion`, i.e what type
+    of entity, is required to display the selection view, using the
+    `getItemSelectorCriterionClass` method.
+
+    In the image provider example, the `ImageItemSelectorCriterion.class` is
+    returned, so the criterion is declared this way:
+
+        public Class<ImageItemSelectorCriterion> getItemSelectorCriterionClass() 
+        {
+            return ImageItemSelectorCriterion.class;
+        }
+
+3.  Declare a method to retrieve the servlet context:
+
+        public ServletContext getServletContext() {
+            return _servletContext;
+        }
+
+    You'll use this later on to help render the markup for your view.
+
+4.  Next, Indicate what information you are returning when the entity is
+    selected, using the `getSupportedItemSelectorReturnTypes` method.
+
+    In the example `URLItemSelectorReturnType` is used, but you could use more
+    return types if the view could return them. More return types means that the
+    view is more reusable:
+
+        public List<ItemSelectorReturnType> getSupportedItemSelectorReturnTypes(
+        ) {
+            return _supportedItemSelectorReturnTypes;
+        }
+
+    This returns the supported Item Selector return types for the view. The
+    return types are added to the list further down in the code. We suggest to
+    follow this same pattern to add your return types to the list:
+
+        private static final List<ItemSelectorReturnType>
+            _supportedItemSelectorReturnTypes =
+            Collections.unmodifiableList(
+                ListUtil.fromArray(
+                    new ItemSelectorReturnType[] {
+                        new URLItemSelectorReturnType()
+                    }));
+
+5.  Next, configure the title, search options, and visibility settings for the
+    selection view:
+
+    For example, the sample selector view has the following settings:
+
+        public String getTitle(Locale locale) {
+            return "Sample Selector";
+        }
+
+        public boolean isShowSearch() {
+            return false;
+        }
```

Comment:
Hey @jhinkey we can write another tutorial about it, but I think it's just as easy as returning true in that method, and then in the ItemSelectorView there's a method renderHTML that indicates where the user performed a search or not based on the `search` parameter. 

	@Override
	public void renderHTML(
			ServletRequest request, ServletResponse response, T t,
			PortletURL portletURL, String itemSelectedEventName, boolean search)
		throws IOException, ServletException {

Then, the keywords the user searched can be obtained as follows:

`String keywords = ParamUtil.getString(request, "keywords");`

But if you think it's interesting we can write a tutorial about it.